nservicebus/pipeline/manipulate-with-behaviors_registernew_core_[6.0,).partial.md

```diff
@@ -7,4 +7,4 @@ Behaviors can also be registered from a `Feature` as shown below:
 
 snippet: RegisterBehaviorFromFeature
 
-WARNING: Behaviors are only created once and the same instance is reused on every invocation of the pipeline. Consequently, every behavior dependency will also behave as a singleton, even if a different option was specified when registering it in [dependency injection](/nservicebus/dependency-injection/). Furthermore, the behavior, and all dependencies called during the invocation phase, need to be concurrency safe.
+WARNING: Behaviors are only created once and the same instance is reused on every invocation of the pipeline. Consequently, every behavior dependency will also behave as a singleton, even if a different option was specified when registering it in [dependency injection](/nservicebus/dependency-injection/). Furthermore, the behavior, and all dependencies called during the invocation phase, need to be concurrency safe and possibly stateless. Storing state in a behavior instance will cause that state to be shared across all message handling sessions, with the risk of undesirable and unwanted side effects.
```

Comment:
I tried to avoid the word Thread safe because in a pure Task based world it is possible that the same thread reenters the behavior multiple times. That's why I went with concurrency safe. I'm rather picky in my messaging because I wanted to untrain developers to think in threads. @kbaley suggestion still adheres to that without using the concurrency safe so I like it!